packages/dtslint-runner/src/prepareAllPackages.ts

```diff
@@ -1,52 +1,31 @@
-import fs from "fs";
-import { pathExists } from "fs-extra";
-import { joinPaths, nAtATime, execAndThrowErrors, npmInstallFlags } from "@definitelytyped/utils";
+import {
+  AllPackages,
+  getDefinitelyTyped,
+  checkParseResults,
+  parseDefinitions
+} from "@definitelytyped/definitions-parser";
+import { joinPaths, loggerWithErrors } from "@definitelytyped/utils";
+import { installDependencies } from "./prepareAffectedPackages";
 import { PreparePackagesOptions, PreparePackagesResult } from "./types";
 
 export async function prepareAllPackages({
   definitelyTypedPath,
-  noInstall
+  noInstall,
+  nProcesses
 }: PreparePackagesOptions): Promise<PreparePackagesResult> {
   const typesPath = joinPaths(definitelyTypedPath, "types");
-  const allPackages = await getAllPackages(typesPath);
+  const [log] = loggerWithErrors();
+  const options = {
+    definitelyTypedPath,
+    progress: false,
+    parseInParallel: nProcesses > 1
+  };
+  const dt = await getDefinitelyTyped(options, log);
+  await parseDefinitions(dt, nProcesses ? { definitelyTypedPath, nProcesses } : undefined, log);
```

Comment:
This is expensive and unnecessary when testing all packages, which is why it wasn’t here before.

Reply:
:+1: That line (`parseDefinitions()`) is easy enough to ax, however it looks like it's needed in order to get the `minTypeScriptVersion` of each package?

Reply:
Ah, I see. If that’s the case then it’s probably worth it. From the description I thought you were consolidating just for code reuse, so I wanted to be clear that there’s a pretty big performance tradeoff.

Reply:
I could get the `minTypeScriptVersion` a lot more efficiently than by calling `parseDefinitions()`, but on the other hand testing all packages is already an expensive operation. Like off the top of my head it currently takes about a couple hours, of which +/- 60 seconds is spent in `parseDefinitions()`? How does this optimization pay off?

Reply:
Yeah, you’re right. My bad 👍